infrastructure/helm/tigerbeetle/templates/statefulset.yaml

```diff
@@ -0,0 +1,224 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "tigerbeetle.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    {{- include "tigerbeetle.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.statefulset.replicas }}
+  serviceName: {{ template "tigerbeetle.fullname" . }}
+  updateStrategy: {{- toYaml .Values.statefulset.updateStrategy | nindent 4 }}
+  podManagementPolicy: {{ .Values.statefulset.podManagementPolicy | quote }}
+  selector:
+    matchLabels:
+      {{- include "tigerbeetle.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.statefulset.annotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "tigerbeetle.selectorLabels" . | nindent 8 }}
+        {{- toYaml .Values.statefulset.labels | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "tigerbeetle.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.statefulset.securityContext | nindent 8 }}
+      initContainers:
+        - name: init-tigerbeetle
+          image: "{{ .Values.statefulset.image.repository }}:{{ .Values.statefulset.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.statefulset.image.pullPolicy }}
+          command:
+            - /bin/sh
+            - -c
+            - | 
+              set -ex
+              REPLICA=${HOSTNAME##*-} 
+              FMT_CLUSTER=$(echo $CLUSTER | awk '{ printf("%010d\n", $1) }')
+              FMT_REPLICA=$(echo $REPLICA | awk '{ printf("%03d\n", $1) }')
+              ls /var/lib/tigerbeetle
+              DATA_FILE=/var/lib/tigerbeetle/cluster_${FMT_CLUSTER}_replica_${FMT_REPLICA}.tigerbeetle
+              if [ ! -f "$DATA_FILE" ]; then 
+                /opt/beta-beetle/tigerbeetle init --cluster=$CLUSTER --replica=$REPLICA --directory=/var/lib/tigerbeetle; 
+              fi
+          env:
+            - name: CLUSTER
+              value: "0"
```

Comment:
This should be configurable from the values file

Reply:
fixed